app/src/main/java/com/andrew/tasky/agenda/data/AgendaApiRepositoryImpl.kt

```diff
@@ -10,23 +10,165 @@ import retrofit2.HttpException
 class AgendaApiRepositoryImpl(
     private val api: AgendaApi
 ) : AgendaApiRepository {
-    override suspend fun getAgendaItems(timezone: String, time: Long
+    override suspend fun getAgendaItems(
+        timezone: String,
+        time: Long
     ): AuthResult<List<AgendaItem>> {
-        TODO("Not yet implemented")
+        return try {
+            val response = api.getAgendaItems(timezone = timezone, time = time)
+            val tasks = response.tasks.map { task ->
+                AgendaItem(
+                    apiId = task.id,
+                    type = AgendaItemType.TASK,
+                    isDone = task.isDone,
+                    title = task.title,
+                    description = task.description ?: "",
+                    startDateAndTime = LocalDateTime.ofEpochSecond(task.time, 0, null),
+                    reminderTime = ReminderTimeConversion.toEnum(
+                        startTimeEpochSecond = task.time,
+                        remindAtEpochSecond = task.remindAt
+                    )
+                )
+            }
```

Comment:
This mapping logic is needed very often which is why we usually create mapper functions, like this:
```
fun TaskDto.toTask(): Task {
    return AgendaItem(...)
}
```